benchmark.go

```diff
@@ -39,8 +42,102 @@ import (
 var (
 	errNegativeDuration = errors.New("duration cannot be negative")
 	errNegativeMaxReqs  = errors.New("max requests cannot be negative")
+
+	// using a global _quantiles slice and _percentileMap mainly for ease of testing, and not passing the same array
+	// around to multiple functions
+	_quantiles     = []float64{0.5, 0.9, 0.95, 0.99, 0.999, 0.9995, 1.0}
+	_percentileMap = map[float64]string{
+		0.5:    "P50",
+		0.9:    "P90",
+		0.95:   "P95",
+		0.99:   "P99",
+		0.999:  "P999",
+		0.9995: "P9995",
+		1.0:    "P100",
+	}
 )
 
+// BenchmarkParameters holds values of all benchmark parameters
+type BenchmarkParameters struct {
+	CPUs        int    `json:"cpus"`
+	Connections int    `json:"connections"`
+	Concurrency int    `json:"concurrency"`
+	MaxRequests int    `json:"maxRequests"`
+	MaxDuration string `json:"maxDuration"`
+	MaxRPS      int    `json:"maxRPS"`
+}
+
+// latencyElement stores individual quantile -> latency value mapping
+type latencyElement struct {
+	quantile string
+	latency  string
+}
+
+// LatencyMap stores latency quantile to value mappings, using a slice to preserve specific ordering of elements for serialization.
+type LatencyMap []latencyElement
+
+// Summary stores the benchmarking summary
+type Summary struct {
+	ElapsedTime   string  `json:"elapsedTime"`
```

Comment:
For JSON, It likely makes more sense to emit something like `ElapsedTimeSeconds float64` rather than a string, which is harder to parse by tools